docs/okeps/okep-5193-user-defined-networks.md

```diff
@@ -0,0 +1,1468 @@
+above may need to be tweaked to match on mark in case 2 pods have overlapping IPs, and are both egressing
+the same interface with different Egress IPs. The flow would look something like this:
+
+![Egress IP VRF LGW](../images/egress-ip-vrf-lgw.svg)
+
+##### Egress Firewall
+
+Egress firewall is enforced at the OVN logical switch, and this proposal has no effect on its functionality.
+
+
+##### Egress QoS
+
+Egress QoS is namespace scoped and functions by marking packets at the OVN logical switch, and this proposal has no
+effect on its functionality.
+
+##### Egress Service
+
+Egress service functions similar to Egress IP in local gateway mode, with the exception that all traffic paths go
+through the kernel networking stack. Egress Service also uses IP rules and VRFs in order to match on traffic and forward
+it out the right network (if specified in the CRD). It uses iptables in order to SNAT packets to the load balancer IP.
+Like Egress IP, with user-defined networks there will need to be IP rules with higher precedence to match on packets from
+specific networks and direct them to the right VRF.
+
+##### Multiple External Gateways (MEG)
+
+There will be no support for MEG or pod direct ingress on any network other than the primary, cluster default network.
+Remember, MEG works the same way in local or shared gateway mode, by utilizing the shared gateway path. This support may
+be enhanced later by extending VRFs/networks outside the cluster.
+
+#### Kubernetes Readiness/Liveness Probes
+
+As previously mentioned, Kubelet probes will continue to work. This includes all types of probes such as TCP, HTTP or
+GRPC. Additionally, we want to restrict host networked pods in namespaces that belong to user-defined networks from
+being able to access pods in other networks. For that reason, we need to block host networked pods from being able to
+access pods via the cluster default network. In order to do this, but still allow Kubelet to send probes; the cgroup
+module in iptables will be leveraged. For example:
+
+```
+root@ovn-worker:/# iptables -L -t raw -v
+Chain PREROUTING (policy ACCEPT 6587 packets, 1438K bytes)
+ pkts bytes target     prot opt in     out     source               destination         
+
+Chain OUTPUT (policy ACCEPT 3003 packets, 940K bytes)
+ pkts bytes target     prot opt in     out     source               destination         
+ 3677 1029K ACCEPT     all  --  any    any     anywhere             anywhere             cgroup kubelet.slice/kubelet.service
+    0     0 ACCEPT     all  --  any    any     anywhere             anywhere             ctstate ESTABLISHED
+  564 33840 DROP       all  --  any    any     anywhere             10.244.0.0/16    
+```
+
+From the output we can see that traffic to the pod network ```10.244.0.0/16``` will be dropped by default. However,
+traffic coming from kubelet will be allowed.
+
+#### Host Networked Pods
```

Comment:
I guess we want to keep the design here even if we didn't get to adding tests or support for this.
I am convinced the feature docs should call it out and enhancement can stay a bit outdated I guess